src/data.ml

```diff
@@ -86,14 +86,12 @@ module Entry = struct
     let decode_value string off = V.decode string (off + K.encoded_size)
 
     let encode' key value f =
-      let encoded_key = K.encode key in
-      let encoded_value = V.encode value in
-      if String.length encoded_key <> K.encoded_size then
-        raise (Invalid_size encoded_key);
-      if String.length encoded_value <> V.encoded_size then
-        raise (Invalid_size encoded_value);
-      f encoded_key;
-      f encoded_value
+      K.encode key (fun s ->
+          if String.length s <> K.encoded_size then raise (Invalid_size s)
+          else f s);
+      V.encode value (fun s ->
+          if String.length s <> K.encoded_size then raise (Invalid_size s)
+          else f s)
```

Comment:
I suspect these checks are always going to fail, because `s` is a subencoding of the key/value here, so it's usually shorter. I believe it's fine to just drop these checks.
Can you change the tests so they can see this ? e.g. by using a pair of strings instead of a single string?